engines/python/setup/djl_python/huggingface.py

```diff
@@ -229,15 +228,14 @@ def wrapped_pipeline(inputs, *args, **kwargs):
 
         return wrapped_pipeline
 
-    @staticmethod
-    def wrap_text_generation_pipeline(hf_pipeline):
+    def wrap_text_generation_pipeline(self, hf_pipeline):
 
         def wrapped_pipeline(inputs, *args, **kwargs):
             model = hf_pipeline.model
             tokenizer = hf_pipeline.tokenizer
-            input_tokens = tokenizer(inputs, padding=True,
-                                     return_tensors="pt").to(
-                                         torch.cuda.current_device())
+            input_tokens = tokenizer(inputs, padding=True, return_tensors="pt")
+            if self.device_id >= 0:
+                input_tokens.to(torch.cuda.current_device())
```

Comment:
you can check if cuda current device existense for using GPU or not. So you don't need to rely on device_id in here